Automattic/wp-calypso
# Add domains table header sorting

In the redesigned domain list we'll allow customers to sort the list by clicking on the table header (or with a select box in mobile view).

NOTE: This is not the final PR because I need to see how the domain item redesign is rendering the domain status in order to make the sorting behave properly. However this PR should be good enough for usability and design review.

#### Changes proposed in this Pull Request

* Update the header component to render the header columns and an arrow next to the active sort key
* Show select box on mobile with all the available sort options
* Implement sorting mechanics ( see #57387 )

Here's how the header looks on desktop and mobile:

![header-demo](https://user-images.githubusercontent.com/1355045/138871420-8cbcadc2-748b-41e5-add5-dd868b472d63.gif)

<img width="377" alt="Screenshot 2021-10-26 at 14 36 57" src="https://user-images.githubusercontent.com/1355045/138870858-0e64379c-1eb1-451b-a546-350a09484f95.png">

#### Testing instructions

* Spin up this branch and verify that you see the domains list table header. The header labels should be clickable. The default sort order is by Status but currently ony sorting by Domain ascending/descending works.
